avm/ptn/ai-platform/baseline/main.bicep

```diff
@@ -0,0 +1,331 @@
+}
+
+resource logAnalyticsWorkspace 'Microsoft.OperationalInsights/workspaces@2023-09-01' = {
+  name: 'log-${name}'
+  location: location
+  tags: tags
+}
+
+module keyVault 'br/public:avm/res/key-vault/vault:0.6.2' = {
+  name: '${uniqueString(deployment().name, location)}-key-vault'
+  params: {
+    name: 'kv-${name}'
+    location: location
+    enableTelemetry: enableTelemetry
+    enableRbacAuthorization: true
+    enableVaultForDeployment: false
+    enableVaultForDiskEncryption: false
+    enableVaultForTemplateDeployment: false
+    networkAcls: {
+      defaultAction: 'Deny'
+      bypass: 'AzureServices'
+    }
+    publicNetworkAccess: 'Disabled'
+    enablePurgeProtection: keyVaultEnablePurgeProtection
+    roleAssignments: [
+      {
+        principalId: managedIdentity.properties.principalId
+        roleDefinitionIdOrName: 'Key Vault Administrator'
+      }
+    ]
+    diagnosticSettings: [
+      {
+        workspaceResourceId: logAnalyticsWorkspace.id
+        logCategoriesAndGroups: [
+          {
+            category: 'AuditEvent'
+            enabled: true
+          }
+        ]
+      }
+    ]
+    tags: tags
+  }
+}
+
+module storageAccount 'br/public:avm/res/storage/storage-account:0.9.1' = {
+  name: '${uniqueString(deployment().name, location)}-storage'
+  params: {
+    name: 'st${name}'
+    location: location
+    enableTelemetry: enableTelemetry
+    allowBlobPublicAccess: false
+    allowSharedKeyAccess: storageAccountAllowSharedKeyAccess
+    defaultToOAuthAuthentication: !storageAccountAllowSharedKeyAccess
+    publicNetworkAccess: 'Disabled'
+    networkAcls: {
+      defaultAction: 'Deny'
+      bypass: 'AzureServices'
+    }
+    roleAssignments: [
+      {
+        principalId: managedIdentity.properties.principalId
+        roleDefinitionIdOrName: 'Storage Blob Data Contributor'
+      }
+      {
+        principalId: managedIdentity.properties.principalId
+        roleDefinitionIdOrName: '69566ab7-960f-475b-8e7c-b3118f30c6bd' // Storage File Data Privileged Contributor
+      }
+    ]
+    tags: tags
+  }
+}
+
+module containerRegistry 'br/public:avm/res/container-registry/registry:0.3.1' = {
+  name: '${uniqueString(deployment().name, location)}-container-registry'
+  params: {
+    name: 'cr${name}'
+    acrSku: 'Premium'
+    location: location
+    enableTelemetry: enableTelemetry
+    publicNetworkAccess: 'Disabled'
+    networkRuleBypassOptions: 'AzureServices'
+    zoneRedundancy: 'Enabled'
+    trustPolicyStatus: containerRegistryTrustPolicyStatus
+    tags: tags
+  }
+}
+
+module applicationInsights 'br/public:avm/res/insights/component:0.3.1' = {
+  name: '${uniqueString(deployment().name, location)}-appi'
+  params: {
+    name: 'appi-${name}'
+    location: location
+    kind: 'web'
+    enableTelemetry: enableTelemetry
+    workspaceResourceId: logAnalyticsWorkspace.id
+    tags: tags
+  }
+}
+
+resource resourceGroup_roleAssignment 'Microsoft.Authorization/roleAssignments@2022-04-01' = {
```

Comment:
Why does the managed identity need to be contributor on the resource group (just out of interest?)

Reply:
Honestly, I'm not sure

I started with just the permissions [here](https://learn.microsoft.com/en-us/azure/machine-learning/how-to-identity-based-service-authentication?view=azureml-api-2&tabs=cli#workspace) but then ran into issues, so in the end I just went with the same as what you would get if you used system assigned managed identity.

There's also an official template [here](https://github.com/Azure/azure-quickstart-templates/tree/master/quickstarts/microsoft.machinelearningservices/machine-learning-dependencies-role-assignment) that you can deploy if you're using user assigned managed identity, and it has Contributor on the resource group

Reply:
Just had a look now, I can see that the identity needs to create CNAME records in the private DNS zone, not sure if there's anything else?

Reply:
Do you think I should scope 'Contributor' to just the resources we're creating? Then, if we later run into issues we can add the missing permissions?

Reply:
I would follow the least privilege approach - Using Contributor is not great IMHO

Reply:
I've taken Contributor off the RG and put it on the resources, let's see how it goes